### PR TITLE
[fmoe] Add EP Support to Two-Stage MoE Op Tests

### PR DESCRIPTION
**Pull request by BangBOOM**
## Motivation

The previous unit test only covered the common non-EP path. Add an EP mode that routes over global experts, simulates rank0 local expert ownership, pads local tokens back to the original shape, and passes num_local_tokens for the fused_moe path.

cc @Duyi-Wang 

## Technical Details

- Add an `ep` option to the legacy two-stage MoE UT path.
- Treat `--expert` as the number of local experts when `ep > 1`.
- Generate routing scores with shape `[token, expert * ep]` so top-k routing can select global experts.
- Simulate rank0 ownership by setting `expert_mask[:expert] = 1` and masking all remote experts through the existing `fused_moe` EP path.
- Filter tokens that have at least one rank0-local expert hit.
- Pad the filtered `input`, `topk_weights`, and `topk_ids` tensors back to the original token shape before calling `fused_moe`.
- Pass `num_local_tokens` to preserve the actual number of valid local tokens.
- Compare only the valid local-token portion of the output for EP cases.
- Preserve existing non-EP behavior when `ep == 1`.
## Test Plan
MI355
```
 python3 op_tests/test_moe_2stage.py \
                              --no-flydsl-csv \
                              -q 4 \
                              -d bf16 \
                              -dim 7168,2048 \
                              -t 128 256 512 \
                              -e 32 \
                              --ep 8 \
                              -k 8 \
                              -a silu \
                              -s f
```

## Test Result

```
| dtype          |   token |   model_dim |   inter_dim |   E |   topk |   actType | gateMode   |   qType | AQDType                | WQDType                | use_g1u1   | doweight_stage1   | strict_accuracy   | check_aot_cache   |   ep | preshuffle   |   hidden_pad |   intermediate_pad |   swiglu_limit |      us |   logits_diff | model   |
|:---------------|--------:|------------:|------------:|----:|-------:|----------:|:-----------|--------:|:-----------------------|:-----------------------|:-----------|:------------------|:------------------|:------------------|-----:|:-------------|-------------:|-------------------:|---------------:|--------:|--------------:|:--------|
| torch.bfloat16 |     128 |        7168 |        2048 |  32 |      8 |         0 | separated  |       3 | torch.float4_e2m1fn_x2 | torch.float4_e2m1fn_x2 | True       | False             | False             | False             |    8 | True         |            0 |                  0 |              0 | 580.948 |   2.59266e-06 | legacy  |
| torch.bfloat16 |     256 |        7168 |        2048 |  32 |      8 |         0 | separated  |       3 | torch.float4_e2m1fn_x2 | torch.float4_e2m1fn_x2 | True       | False             | False             | False             |    8 | True         |            0 |                  0 |              0 | 578.488 |   3.0777e-06  | legacy  |
| torch.bfloat16 |     512 |        7168 |        2048 |  32 |      8 |         0 | separated  |       3 | torch.float4_e2m1fn_x2 | torch.float4_e2m1fn_x2 | True       | False             | False             | False             |    8 | True         |            0 |                  0 |              0 | 562.167 |   2.73326e-06 | legacy  |
```

## Submission Checklist

- [ ] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

### CODE REVIEW

**Comment by zufayu** [comment]
@BangBOOM This PR has been open since June 2026 with merge conflicts and no review activity since. Is the EP support for two-stage MoE op tests still something you plan to pursue? If yes, please rebase onto current main. If there's no update within ~2 weeks, we'll close this to keep the PR queue clean — it can always be reopened or resubmitted.
